x-pack/plugins/security_solution/public/flyout/preview/context.tsx

```diff
@@ -52,12 +60,21 @@ export const PreviewPanelProvider = ({
   ruleId,
   children,
 }: PreviewPanelProviderProps) => {
+  const currentSpaceId = useSpaceId();
+  const eventIndex = indexName ? getAlertIndexAlias(indexName, currentSpaceId) ?? indexName : '';
```

Comment:
Just as a quick note, this `getAlertIndexAlias` function is going to be a bad idea in the long run. It was code that I added a little while ago before knowing the `_alias` api existed: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-get-alias.html which is what we probably should use. It'll delay the loading of the flyout a little bit longer to make that request, but is definitely safer in the long run. This was added to get around the fact that the concrete index on the alert `.internal-[indexHere]` isn't accessible except through it's alias `.alerts-security...`

Reply:
thanks @michaelolo24 I remember you mentioning this in the past. I've created [a ticket](https://github.com/elastic/security-team/issues/7341) for it. For now the usage done here is similar to the other 2 context files, and I feel like updating the code would be outside of the scope of this PR.
We will tackle this in BC period, or most likely for `8.11` release, if that is ok!